lib/src/main/java/com/auth0/android/lock/WebProvider.java

```diff
@@ -28,15 +32,19 @@
     /**
      * Configures a new instance of the WebAuthProvider.Builder and starts it.
      *
-     * @param activity    a valid Activity context
-     * @param connection  to use in the authentication
-     * @param callback    to deliver the authentication result to
-     * @param requestCode to use in the startActivityForResult request
+     * @param activity       a valid Activity context
+     * @param connection     to use in the authentication
+     * @param authParameters extra authentication parameters to use. If null, it will use the ones defined by Options#getAuthenticationParameters
+     * @param callback       to deliver the authentication result to
+     * @param requestCode    to use in the startActivityForResult request
      */
-    public void start(Activity activity, String connection, AuthCallback callback, int requestCode) {
+    public void start(@NonNull Activity activity, @NonNull String connection, @Nullable Map<String, Object> authParameters, @NonNull AuthCallback callback, int requestCode) {
```

Comment:
The `WebProvider` class it's just an internal (pkg priv) wrapper for calls to the auth0's `WebAuthProvider`. It was created with the purpose of passing just the `Options` in the constructor and letting it handle all the params settings inside, as they weren't going to change at least for Lock's lifecycle.. 
With this PR, I need to set a value that it's different everytime the "oauth login" it's called -> the username/email to set as `login_hint`. There were 2 ways to do it:
- Add a setter for this parameter (or a generic setter for any parameter or parameters) and call it before each call to `start`. This would require to keep an extra "map" reference to store this extra values.
- Change the signature (as it's internal) and add these "extra parameters", which will default to the `options.getAuthenticationParameters()` if their value is `null`.

I did the latter, but I can do the first one if that's what you prefer. Do you think of any other options?